x-pack/plugins/security_solution/server/lib/detection_engine/signals/filters/create_set_to_filter_against.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash/fp';
+import { CreateSetToFilterAgainstOptions } from './types';
+
+/**
+ * Creates a field set to filter against using the stringed version of the
+ * data type for compare. Creates a set of list values that are stringified that
+ * are easier to work with as well as ensures that deep values can work since it turns
+ * things into a string using JSON.stringify().
+ *
+ * @param events The events to filter against
+ * @param field The field checking against the list
+ * @param listId The list id for the list function call
+ * @param listType The type of list for the list function call
+ * @param listClient The list client API
+ * @param logger logger for errors, debug, etc...
+ */
+export const createSetToFilterAgainst = async <T>({
+  events,
+  field,
+  listId,
+  listType,
+  listClient,
+  logger,
```

Comment:
Can we also pass in `buildRuleMessage` and use it in the log statements below? I think this would be heplful when debugging later on.

Reply:
Yeah, I can.